Source/WebCore/platform/mediarecorder/MediaRecorderPrivateEncoder.cpp

```diff
@@ -276,12 +277,70 @@ void MediaRecorderPrivateEncoder::appendAudioSampleBuffer(const PlatformAudioDat
     // explicitly allow the following allocation(s).
     DisableMallocRestrictionsForCurrentThreadScope disableMallocRestrictions;
 
+    if (!m_currentAudioSamplingRate) {
```

Comment:
In theory, it should be possible for microphone to change of sample rate.
We could check that m_currentAudioSamplingRate != description.sampleRate().
Though I guess, the compressor might not like the change and this would give the risk of changing the ring buffer.
Maybe, when we detect that the description changes, we stop the recording or we at least log an error.

Reply:
testing has indicated that the sampling rate can't change; if it did quite a few audio listener would be wrong.
If you start with a stream at 24kHz and you switch to one at 48, it would be resampled to 24 before getting there.

The assumption that the rate never changed has been there ever since the MediaRecorder was added, why worry about it now ?

Reply:
> > testing has indicated that the sampling rate can't change
> 
> In that case an ASSERT can't hurt

it's there already, in `MediaRecorderPrivateEncoder::appendAudioSampleBuffer` `ASSERT(!m_currentAudioSamplingRate || m_currentAudioSamplingRate == description.sampleRate(), "sampling rate should never change over a session");`

Reply:
I added a detection for sampling rate change, which will abort the recorder and force writing the data already obtained.

I will take on a follow up fix that make use of a resampler if needed.

Reply:
for the record, taking this offline with @youennf I could produce a case where the sampleRate changed, I first had to enable the settings to have the capture in the GPU process and then start recording and in the middle of the session do: ```
setTimeout(() => {
            stream.getAudioTracks()[0].applyConstraints({sampleRate:44100})
        }, 1500);
```
it changed from 48kHz to 44.1kHz

Reply:
will follow up in https://bugs.webkit.org/show_bug.cgi?id=282489